maze_transformer/evaluation/path_evals.py

```diff
@@ -0,0 +1,109 @@
+from typing import Callable, Iterable, TypeAlias
+
+import numpy as np
+from muutils.tensor_utils import NDArray
+
+from maze_transformer.generation.constants import Coord, CoordTup
+from maze_transformer.generation.latticemaze import LatticeMaze
+
+# pylint: disable=unused-argument
+MazePath: TypeAlias = NDArray["node x_y_pos", int]
+PathEvalFunction = Callable[[LatticeMaze, MazePath, MazePath], float]
+
+
+def path_as_segments_iter(path: MazePath) -> Iterable[tuple]:
+    """
+    Iterate over the segments of a path.
+    """
+    i: int
+    n_s: Coord | CoordTup
+    n_e: Coord | CoordTup
+    for i, n_s in enumerate(path[:-1]):
+        n_e = path[i + 1]
+        yield (n_s, n_e)
+
+
+class PathEvals:
+    """array path based eval functions. first path is always the "ground truth" path"""
+
+    @staticmethod
+    def node_overlap(
+        maze: LatticeMaze, solution: MazePath, prediction: MazePath, /
```

Comment:
```suggestion
        solution: MazePath, prediction: MazePath, /
```

Reply:
Yeah all these suggestions to remove unused args won't work unless we significantly change the approach here. The path evals all need to have the same interface because they're called in a for loop with the same inputs.